.github/workflows/build_and_run_apps.yml

```diff
@@ -0,0 +1,143 @@
+name: Build and Run Apps
+
+on:
+  schedule:
+    - cron: '0 0 * * *' # Once per day at midnight
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  build:
+    name: Build Apps
+    strategy:
+      fail-fast: false
+      matrix:
+        idf_ver: 
+          - "release-v5.0"
+          - "release-v5.1"
+          - "release-v5.2"
+          - "release-v5.3"
+          - "latest"
+        parallel_index: [1,2,3,4,5] # Update --parallel-count below when changing this
+    runs-on: ubuntu-22.04
+    container: espressif/idf:${{ matrix.idf_ver }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'true'
+      - name: Install dependencies
+        shell: bash
+        run: |
+          . ${IDF_PATH}/export.sh
+          pip install idf-component-manager idf-build-apps --upgrade
+      - name: Fix git repo permissions
+        # Needed by the next git diff step.
+        # See https://github.com/actions/runner/issues/2033
+        if: github.event_name == 'pull_request'
+        run: |
+          build_dir=$PWD
+          cd /
+          git config --global --add safe.directory $build_dir
+          cd -
+      - name: Find files changed in PR
+        if: github.event_name == 'pull_request'
+        run: |
+          git fetch --recurse-submodules=no origin ${{ github.base_ref }}:base_ref
+          git fetch --recurse-submodules=no origin pull/${{ github.event.pull_request.number }}/head:pr_ref
+          git diff --name-only -r base_ref pr_ref > changed_files.txt
+          python3 .github/get_pr_changes.py -v changed_files.txt idf_build_apps_args.txt
```

Comment:
I did not check the python script fully, but FYI, this is how we get list of changed files in BSP https://github.com/espressif/esp-bsp/blob/master/.github/workflows/build-applications.yml#L61

Reply:
Thanks, I have considered using tj-actions/changed-files, but in the end decided that 3 lines (git fetch, git fetch, git diff) were easier to maintain than a dependency on another workflow.

The `get_pr_changes` is probably badly named — it takes the list of changed files and converts it into the list of changed components in the format of idf_build_apps arguments. I guess I'll rename the script to make this more clear.